internal/system/snapd_client.go

```diff
@@ -0,0 +1,155 @@
+package system
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+const (
+	// StatusActive represents an active snap installation.
+	StatusActive = "active"
+)
+
+// SnapdClient is a minimal client for the snapd REST API.
+type SnapdClient struct {
+	httpClient *http.Client
+}
+
+// NewSnapdClient creates a new snapd API client.
+func NewSnapdClient() *SnapdClient {
+	return &SnapdClient{
+		httpClient: &http.Client{
+			Transport: &http.Transport{
+				DialContext: func(ctx context.Context, _, _ string) (net.Conn, error) {
+					return (&net.Dialer{
+						Timeout:   30 * time.Second,
+						KeepAlive: 30 * time.Second,
+					}).DialContext(ctx, "unix", "/run/snapd.socket")
```

Comment:
This would be clearer as:
```suggestion
					dialer := net.Dialer{
						Timeout:   30 * time.Second,
						KeepAlive: 30 * time.Second,
					}
					return dialer.DialContext(ctx, "unix", "/run/snapd.socket")
```